engine/lib/ast_utils.ml

```diff
@@ -777,6 +792,9 @@ module Make (F : Features.T) = struct
 
   let make_wild_pat (typ : ty) (span : span) : pat = { p = PWild; span; typ }
 
+  let make_unguarded_arm (arm_pat : pat) (body : expr) (span : span) : arm =
+    { arm = { arm_pat; body; guard = None }; span }
```

Comment:
Can you make this into a `make_arm`, with an optional (defaulting to `None`) `?guard` argument?